hs_tracking/middleware.py

```diff
@@ -2,17 +2,43 @@
 import utils
 import re
 
-RESOURCE_RE = re.compile('resource/([0-9a-f]{32})/')
+RESOURCE_RE = re.compile('resource/([0-9a-f]{32})/')  # parser for resource id
+BAG_RE = re.compile('bags/([0-9a-f]{32})\.zip')  # parser for resource id
+LANDING_RE = re.compile('resource/([0-9a-f]{32})/$')  # reference to resource home page
+REST_RE = re.compile('/hsapi/')  # reference to REST or internal
+INTERNAL_RE = re.compile('/hsapi/_internal/')  # reference to an internal page
 
 
-def get_resource_id_from_url(request):
+def get_resource_id_from_url(path):
+    """ read a resource id from a URL """
+    m = RESOURCE_RE.search(path)
+    if m and m.group(1):
+        return m.group(1)
+    m = BAG_RE.search(path)
+    if m and m.group(1):
+        return m.group(1)
+    return None
 
-    m = RESOURCE_RE.search(request)
-    if (m and m.group(1)):
-        resource_id = m.group(1)
-        return resource_id
+
+def get_rest_from_url(path):
+    """ determine whether a URL is a REST call or not """
+    m = REST_RE.search(path)
+    if m:
+        n = INTERNAL_RE.search(path)
+        if n:
+            return False
+        else:
+            return True
+    else:
+        return False
+
```

Comment:
how about:
match_restful = REST_RE.search(path)
match_internal = INTERNAL_RE.search(path)
return match_restful and match_internal

Reply:
Not good enough because the variable requires a real bool and not a search result. 
See current version.